packages/frontend/src/pages/campaigns/index.tsx

```diff
@@ -88,25 +38,60 @@ export const Campaigns = () => {
     const { loading, kpiTokens: searchedResolvedKPITokens } =
         useSearchedResolvedKPITokens(debouncedSearchQuery);
 
-    //  show filters
-    const [filtersOpen, setFilterOpen] = useState(false);
+    //  toggle filters
     const toggleFilters = () => setFilterOpen(!filtersOpen);
+    const [filtersOpen, setFilterOpen] = useState(false);
 
     // page settings
     const [pageSize, setPageSize] = useState(12);
     const [ordering, setOrdering] = useState<SelectOption>(ORDERING_OPTIONS[0]);
+
+    // filters
     const [campaignState, setCampaignState] = useState<SelectOption>(
         STATE_OPTIONS[0]
     );
+    // side filters
+    const [selectedTemplates, setSelectedTemplates] = useState<Set<string>>(
+        new Set<string>()
+    );
+
+    const [selectedOracles, setSelectedOracles] = useState<Set<string>>(
+        new Set<string>()
+    );
+    const handleSelectedTemplatesUpdate = useCallback(
+        (newSelectedTemplates: Set<string>) =>
+            setSelectedTemplates(new Set(newSelectedTemplates)),
+        []
+    );
+    const handleSelectedOraclesTemplatesUpdate = useCallback(
+        (newSelectedOracles: Set<string>) =>
+            setSelectedOracles(new Set(newSelectedOracles)),
+        []
+    );
 
     // filter results
-    const filteredTokens = useMemo(() => {
+    const filteredKPITokensByState = useMemo(() => {
         return filterResolvedKPITokens(
             searchedResolvedKPITokens,
             campaignState.value as unknown as CampaignState
         );
     }, [searchedResolvedKPITokens, campaignState]);
 
+    const filteredTokens = useMemo(() => {
+        const noFiltersSelected =
+            selectedTemplates.size === 0 && selectedOracles.size === 0;
+
+        if (noFiltersSelected) return filteredKPITokensByState;
```

Comment:
I think we can remove `noFiltersSelected` and put the condition directly in the `if`.